mm2src/coins/lp_coins.rs

```diff
@@ -4489,6 +4479,42 @@ pub async fn lp_coinfind_any(ctx: &MmArc, ticker: &str) -> Result<Option<MmCoinS
     Ok(coins.get(ticker).cloned())
 }
 
+/// Finds unique account IDs with any status (active or inactive).
+pub async fn find_unique_account_ids_any(ctx: &MmArc) -> Result<HashSet<String>, String> {
+    find_unique_account_ids(ctx, false).await
+}
+
+/// Finds unique account IDs for active accounts.
+pub async fn find_unique_account_ids_active(ctx: &MmArc) -> Result<HashSet<String>, String> {
+    find_unique_account_ids(ctx, true).await
+}
+
+/// Finds unique account IDs based on the given context and active status.
+async fn find_unique_account_ids(ctx: &MmArc, active_only: bool) -> Result<HashSet<String>, String> {
+    // Using a HashSet to ensure uniqueness efficiently
+    // Initialize with default wallet pubkey as coin.account_db_id() will return None by default.
+    let mut account_ids = HashSet::from([ctx.rmd160_hex()]);
+
+    let coin_ctx = try_s!(CoinsContext::from_ctx(ctx));
+    let coins = coin_ctx.coins.lock().await;
+
+    for coin in coins.values() {
```

Comment:
To iterate only once:

```suggestion
    for (_, coin) in coins.deref() {
```